src/dynsight/_internal/analysis/shannon_entropy.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import annotations
+
+from pathlib import Path
+
+import numpy as np
+import numpy.typing as npt
+
+
+def compute_data_entropy(
+    data: npt.NDArray[np.float64],
+    data_range: tuple[float, float],
+    n_bins: int,
+) -> float:
+    """Compute the entropy of a data distribution.
+
+    It is normalized so that a uniform distribution has unitary entropy.
+
+    * Original author: Matteo Becchi
+
+    Parameters:
+        data : the dataset of which the entropy has to be computed.
```

Comment:
Change to formatting like this:

```python
        data:
            COMMENT.
```

Removing the typing information.

Reply:
Done